### PR TITLE
feat(output): open output for last test run

### PR DESCRIPTION
**Pull request by marcelbeumer**
Running `:lua require('neotest').output.open({ last_run = true })` will open output of last test run. 

I created this to support the workflow:

- Run a test with `:lua require("neotest").run.run()`.
- Edit implementation in a split, next to the test file.
- Run `:lua require("neotest").run.run_last()` to run last test while still in non-test buffer.
- Run `:lua require('neotest').output.open({ last_run = true })` to check it's output still in the non-test buffer.

Should not be feature creep, so if there's a better way to support this workflow, let me know!

### CODE REVIEW

File: lua/neotest/consumers/output.lua [hunk 1]
```diff
@@ -171,7 +173,15 @@ function neotest.output.open(opts)
   end
   async.run(function()
     local tree, adapter_id
-    if not opts.position_id then
+    if opts.last_run then
+      local last_run = run.get_last_run()
```
Comment (rcarriga) [inline lua/neotest/consumers/output.lua:177]: Need to check if `last_run` is `nil`

File: lua/neotest/consumers/run.lua [hunk 2]
```diff
@@ -13,6 +13,11 @@ local neotest = {}
 --- A consumer providing a simple interface to run tests.
 neotest.run = {}
 
+---@private
```
Comment (rcarriga) [inline lua/neotest/consumers/run.lua:16]: Similar comment, we can't treat this as private if another consumer is using it.

I think we should do the unpacking (and thus nil check) here as it'll be a more consistent API. 

Please also add a doc message with a description and return values, the Github Actions will generate a commit to update the docs with it.

**Review by rcarriga — CHANGES_REQUESTED** [review]
Thanks for the PR! I definitely think this is a good feature to have :smile: Just a few small comments

File: lua/neotest/consumers/run.lua (hunk 2 above)
Comment (marcelbeumer) [reply lua/neotest/consumers/run.lua]: Done. Still seems a bit strange to have a public `.run()` that returns a `tree, adapter_id`?

File: lua/neotest/consumers/run.lua [hunk 3]
```diff
@@ -197,6 +197,16 @@ function neotest.run.adapters()
   return client:get_adapters()
 end
 
+--- Get last test run tree and adapter id.
+---@return neotest.Tree | nil, integer | nil
+function neotest.run.get_last_run()
```
Comment (marcelbeumer) [inline lua/neotest/consumers/run.lua:202]: @rcarriga: still seems a bit strange to have a public `.run()` that returns a `tree, adapter_id`? But have to admit I didn't spend enough time to really understand the code's data structures and API's properly..

Comment (rcarriga) [reply lua/neotest/consumers/run.lua:202]: Sorry I should have been clearer :facepalm: The function just return `position_id, last_args`, not the tree as yes you're right the tree is not really meant for general usage, just for consumers.

File: lua/neotest/consumers/output.lua [hunk 4]
```diff
@@ -171,7 +172,16 @@ function neotest.output.open(opts)
   end
   async.run(function()
     local tree, adapter_id
-    if not opts.position_id then
+    if opts.last_run then
+      local position_id, last_args = require("neotest").run.get_last_run()
```
Comment (marcelbeumer) [inline lua/neotest/consumers/output.lua:176]: when doing a `require('neotest').run` on top of the file of a `neotest.run` here, I will get an error run being `nil`

**Comment by rcarriga** [comment]
LGTM, thanks again for taking this on :grin:

**Pull request merged**